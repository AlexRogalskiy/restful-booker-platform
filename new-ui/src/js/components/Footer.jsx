import React from "react";

export default class Footer extends React.Component {

    constructor() {
        super();
    }

    render() {
      return(<footer className="footer">
                <div className="container">
                    <p className="text-muted">restful-booker-platform v1.0 - Created by Mark Winteringham - &copy; 2016</p>
                </div>
            </footer>);
    }
  }