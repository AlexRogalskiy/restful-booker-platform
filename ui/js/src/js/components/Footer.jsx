import React from "react";
import { Link } from 'react-router-dom';

export default class Footer extends React.Component {

    constructor() {
        super();
    }

    render() {
      return(
            <footer id="footer" className="footer">
                <div className="container">
                    <p className="text-muted">restful-booker-platform v3.0 - Created by <a href="http://www.mwtestconsultancy.co.uk">Mark Winteringham</a> - &copy; 2018 <Link to={'/cookiepolicy'}>Cookie-Policy</Link> <span style={{float : "right"}}>Learn more about <a href="https://automationintesting.com" alt="Automation in testing site">Automation in Testing</a> here</span></p>
                </div>
            </footer>);
    }
  }