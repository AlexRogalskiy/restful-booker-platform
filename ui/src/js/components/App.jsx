import React from 'react';
import { Route, Switch } from 'react-router-dom'
import RoomListings from './RoomListings.jsx';
import Search from './Search.jsx';
import Nav from './Nav.jsx';
import Login from './Login.jsx'
import RoomDetails from './RoomDetails.jsx';
import Report from './Report.jsx';
import Cookies from 'universal-cookie';
import { API_ROOT } from '../api-config';

export default class App extends React.Component {

    constructor() {
        super();

        this.state = {
            isAuthenticated : false
        }
        
        this.setAuthenticate = this.setAuthenticate.bind(this);
    }

    componentDidMount(){
        const cookies = new Cookies();

        fetch(API_ROOT.auth + '/validate', {
                method: 'POST',
                headers: {
                    'Accept': 'application/json',
                    'Content-Type': 'application/json'
                },
                body : JSON.stringify({ token: cookies.get('token')})
            })
            .then(res => {
                if(res.status == 200){
                    this.setState({isAuthenticated: true});
                }
            })
    }

    setAuthenticate(e){
        this.setState({
            isAuthenticated : e
        });
    }

    render() {
        let app = null;

        if(!this.state.isAuthenticated){
            app = <Login setAuthenticate={this.setAuthenticate}/>
        } else {
            app = <div>
                    <Nav setAuthenticate={this.setAuthenticate} />
                    <Switch>
                        <Route exact path='/' render={(props) => <RoomListings {...props} />} />
                        <Route exact path='/search' component={Search} {...this.props}/>
                        <Route exact path='/room/:id' render={({ location, match }) => (
                            <RoomDetails isAuthenticated={this.state.isAuthenticated} params={match.params}/>
                        )} />
                        <Route exact path='/report' component={Report} />
                    </Switch>
                </div>
        }

        return(
            <div className="container">
                {app}
            </div>
        );
    }
}
