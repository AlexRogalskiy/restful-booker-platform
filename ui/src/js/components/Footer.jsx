import React from "react";

export default class Footer extends React.Component {

    constructor() {
        super();
    }

    render() {
      return(<footer className="footer">
                <div className="container">
                    <p className="text-muted">restful-booker-platform v3.0 - Created by <a href="http://www.mwtestconsultancy.co.uk">Mark Winteringham</a> - &copy; 2018</p>
                </div>
            </footer>);
    }
  }