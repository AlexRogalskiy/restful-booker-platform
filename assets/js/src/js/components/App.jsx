import React from 'react';
import { HashRouter, Route, Switch } from 'react-router-dom'
import { SHOW_WELCOME } from '../api-config';
import ReactModal from 'react-modal';
import Cookies from 'universal-cookie';
import { createBrowserHistory as createHistory } from 'history'
import ReactGA from 'react-ga';

import AdminContainer from './AdminContainer.jsx';
import Welcome from './Welcome.jsx';
import CookiePolicy from './CookiePolicy.jsx';
import PrivacyPolicy from './PrivacyPolicy.jsx';
import Home from './Home.jsx';
import Footer from './Footer.jsx';

if (process.env.NODE_ENV !== 'test') ReactModal.setAppElement('#root');

const history = createHistory();

ReactGA.initialize('UA-118712228-3');

history.listen((location) => {
    ReactGA.pageview(location.pathname + location.hash);
});

export default class App extends React.Component {

    constructor() {
        super();

        this.state = {
            showWelcome : false
        }
        
        this.setWelcome = this.setWelcome.bind(this);

        ReactGA.pageview(window.location.pathname + window.location.hash);
    }

    componentDidMount(){
        const cookies = new Cookies();

        if(typeof cookies.get('welcome') === 'undefined' && SHOW_WELCOME){
           this.setState({
                showWelcome : true
            })
        }
    }

    setWelcome(e){
        this.setState({
            showWelcome : e
        });
    }

    render() {
        let welcome = null;

        if(this.state.showWelcome){
            welcome = <div>
                <Welcome setWelcome={this.setWelcome} />
            </div>
        }

        return(
            <HashRouter>
                {welcome}
                    <Switch>
                        <Route path='/admin/' render={() => (
                            <AdminContainer />
                        )} />
                        <Route exact path='/' component={Home} />
                        <Route exact path='/cookie' component={CookiePolicy} />
                        <Route exact path='/privacy' component={PrivacyPolicy} />
                    </Switch>
                <Footer />
            </HashRouter>
        );
    }
}
