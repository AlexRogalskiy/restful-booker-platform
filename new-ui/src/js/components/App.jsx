import React from "react";
import { Switch, Route, Link } from 'react-router-dom'
import Hotel from './Hotel.jsx';
import Search from './Search.jsx';
import Footer from './Footer.jsx';
import Nav from './Nav.jsx';

export default class App extends React.Component {

    constructor() {
        super();
    }

    render() {
      return(
          <div className="container">
            <Nav />
            <Link to="hotel">Hotel</Link>
            <Link to="search">Search</Link>
            <Switch>
              <Route exact path='/hotel' component={Hotel}/>
              <Route exact path='/search' component={Search}/>
            </Switch>
          </div>
      );
    }
  }

  