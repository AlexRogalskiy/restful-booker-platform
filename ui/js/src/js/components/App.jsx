import React, { Suspense } from 'react';
import { Route, Switch } from 'react-router-dom'
import { API_ROOT, SHOW_WELCOME } from '../api-config';

import Nav from './Nav.jsx';
import Login from './Login.jsx'
import Report from './Report.jsx';
import Welcome from './Welcome.jsx';
import Cookies from 'universal-cookie';
import CookiePolicy from './CookiePolicy.jsx';
import PrivacyPolicy from './PrivacyPolicy.jsx';
import Loading from './Loading.jsx';
import Home from './Home.jsx';
import Branding from './Branding.jsx';
import Footer from './Footer.jsx';
import ReactModal from 'react-modal';

const RoomListings = React.lazy(() => import('./RoomListings.jsx'));
const RoomDetails = React.lazy(() => import('./RoomDetails.jsx'));

ReactModal.setAppElement('#root');

export default class App extends React.Component {

    constructor() {
        super();

        this.state = {
            isAuthenticated : false,
            showWelcome : false
        }
        
        this.setAuthenticate = this.setAuthenticate.bind(this);
        this.setWelcome = this.setWelcome.bind(this);
    }

    componentDidMount(){
        const cookies = new Cookies();

        if(typeof cookies.get('welcome') === 'undefined' && SHOW_WELCOME){
           this.setState({
                isAuthenticated : this.state.isAuthenticated,
                showWelcome : true
            })
        }

        fetch(API_ROOT.auth + '/validate', {
                method: 'POST',
                headers: {
                    'Accept': 'application/json',
                    'Content-Type': 'application/json'
                },
                body : JSON.stringify({ token: cookies.get('token')})
            })
            .then(res => {
                if(res.status == 200){
                    this.setAuthenticate(true);
                }
            })
    }

    setAuthenticate(e){
        this.setState({
            isAuthenticated : e,
            showWelcome : this.state.showWelcome
        });
    }

    setWelcome(e){
        this.setState({
            isAuthenticated : this.state.isAuthenticated,
            showWelcome : e
        });
    }

    render() {
        let welcome = null;

        if(this.state.showWelcome){
            welcome = <div>
                <Welcome setWelcome={this.setWelcome} />
            </div>
        }

        return(
            <div>
                {welcome}
                    <Switch>
                        <Route path='/admin/' render={() => (
                            <div>
                                <Nav setAuthenticate={this.setAuthenticate} isAuthenticated={this.state.isAuthenticated} />
                                <div className="container">
                                    <div>
                                        {this.state.isAuthenticated ? (
                                            <div>
                                                <Route exact path='/admin/' render={(props) => (
                                                    <div>
                                                        <Suspense fallback={<Loading />}>
                                                            <RoomListings {...props} />
                                                        </Suspense>
                                                    </div>
                                                )} />
                                                <Route exact path='/admin/room/:id' render={({ location, match }) => (
                                                    <div>
                                                        <Suspense fallback={<Loading />}>
                                                            <RoomDetails params={match.params}/>
                                                        </Suspense>
                                                    </div>
                                                )} />
                                                <Route exact path='/admin/report' component={Report} />
                                                <Route exact path='/admin/branding' component={Branding} />
                                            </div>
                                        ) : (
                                            <div>
                                                <Login setAuthenticate={this.setAuthenticate} />
                                            </div>
                                        )}
                                    </div>
                                </div>
                            </div>
                        )} />
                        <Route exact path='/' component={Home} />
                        <Route exact path='/cookie' component={CookiePolicy} />
                        <Route exact path='/privacy' component={PrivacyPolicy} />
                    </Switch>
                <Footer />
            </div>
        );
    }
}
