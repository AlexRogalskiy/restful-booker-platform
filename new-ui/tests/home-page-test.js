import React from 'react';
import Footer from '../src/js/components/Footer.jsx';
import App from '../src/js/components/App.jsx';

test('App container component', () => {
    const app = shallow(
        <App />
    );

    expect(app).toMatchSnapshot();
});

test('Footer component', () => {
    const footer = shallow(
        <Footer />
    );

    expect(footer).toMatchSnapshot();
});