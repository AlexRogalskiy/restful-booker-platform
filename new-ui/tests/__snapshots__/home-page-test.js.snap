// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container component 1`] = `
<div
  className="container"
>
  <Nav />
  <Link
    replace={false}
    to="hotel"
  >
    Hotel
  </Link>
  <Link
    replace={false}
    to="search"
  >
    Search
  </Link>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/hotel"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/search"
    />
  </Switch>
</div>
`;

exports[`Footer component 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <p
      className="text-muted"
    >
      restful-booker-platform v1.0 - Created by Mark Winteringham - Â© 2016
    </p>
  </div>
</footer>
`;
